eth/downloader/downloader_test.go

```diff
@@ -921,6 +921,58 @@ func testMultiProtoSync(t *testing.T, protocol uint, mode SyncMode) {
 	}
 }
 
+// Tests that if a block is empty (e.g. header only), no body request should be
+// made, and instead the header should be assembled into a whole block in itself.
+func TestEmptyShortCircuit67Full(t *testing.T)  { testEmptyShortCircuit(t, istanbul.Celo67, FullSync) }
+func TestEmptyShortCircuit67Fast(t *testing.T)  { testEmptyShortCircuit(t, istanbul.Celo67, FastSync) }
+func TestEmptyShortCircuit67Light(t *testing.T) { testEmptyShortCircuit(t, istanbul.Celo67, LightSync) }
```

Comment:
Where did these tests come from?

Reply:
Upstream. We removed those before. I restore them